core/cas-server-core-services/src/test/java/org/apereo/cas/services/DomainServicesManagerTests.java

```diff
@@ -0,0 +1,269 @@
+        assertNull(this.domainServicesManager.findServiceBy(1000));
+    }
+
+    @Test
+    public void verifyEvaluationOrderOfServices() {
+        final RegexRegisteredService r = new RegexRegisteredService();
+        r.setId(100);
+        r.setName(TEST);
+        r.setServiceId(TEST);
+        r.setEvaluationOrder(200);
+
+        final RegexRegisteredService r2 = new RegexRegisteredService();
+        r2.setId(101);
+        r2.setName(TEST);
+        r2.setServiceId(TEST);
+        r2.setEvaluationOrder(80);
+
+        final RegexRegisteredService r3 = new RegexRegisteredService();
+        r3.setId(102);
+        r3.setName("Sample test service");
+        r3.setServiceId(TEST);
+        r3.setEvaluationOrder(80);
+
+        this.domainServicesManager.save(r);
+        this.domainServicesManager.save(r3);
+        this.domainServicesManager.save(r2);
+
+        final List<RegisteredService> allServices = new ArrayList<>(this.domainServicesManager.getAllServices());
+
+        //We expect the 3 newly added services, plus the one added in setUp()
+        assertEquals(4, allServices.size());
+
+        assertEquals(allServices.get(0).getId(), r3.getId());
+        assertEquals(allServices.get(1).getId(), r2.getId());
+        assertEquals(allServices.get(2).getId(), r.getId());
+    }
+
+    @Test
+    public void verifyServiceCanBeUpdated() throws Exception {
+        final int serviceId = 2500;
+        final String description = "desc";
+
+        final RegexRegisteredService service = new RegexRegisteredService();
+        service.setId(serviceId);
+        service.setName("serviceName");
+        service.setServiceId("serviceId");
+        service.setEvaluationOrder(1000);
+
+        domainServicesManager.save(service);
+
+        service.setDescription(description);
+
+        domainServicesManager.save(service);
+
+        final Collection<RegisteredService> serviceRetrieved = domainServicesManager.findServiceBy(RegexRegisteredService.class::isInstance);
+
+        assertEquals(description, serviceRetrieved.toArray(new RegisteredService[]{})[0].getDescription());
+    }
+
+    @Test
+    public void verifyServiceIsUpdatedAfterALoad() throws Exception {
+        final int serviceId = 2500;
+        final String description = "desc";
+
+        final RegexRegisteredService service = new RegexRegisteredService();
+        service.setId(serviceId);
+        service.setName("serviceName");
+        service.setServiceId("serviceId");
+        service.setEvaluationOrder(1000);
+        dao.save(service);
+        service.setDescription(description);
+
+        dao.save(service);
+        domainServicesManager.load();
+
+        final Collection<RegisteredService> serviceRetrieved = domainServicesManager.findServiceBy(RegexRegisteredService.class::isInstance);
+
+        assertEquals(description, serviceRetrieved.toArray(new RegisteredService[]{})[0].getDescription());
+    }
+
+    private static class SimpleService implements Service {
```

Comment:
I haven't checked honestly, but you might be able to delete this class entirely and use `WebUtils.getService()` or something similar in RegisteredServicesTestUtils, etc.